### PR TITLE
feat: error code urls.

### PR DESCRIPTION
**Pull request by thantos**
Related to #220 

Automatically add functionless.org error code urls (auto generated) to synth errors with codes.

> SynthError: Expected input function to AppsyncResolver to be compiled by Functionless. Make sure you have the Functionless compiler plugin configured correctly.
>
> https://functionless.org/docs/error-codes#function-not-compiled-by-functionless-plugin

### CODE REVIEW

File: src/error-code.ts [hunk 1]
```diff
@@ -3,10 +3,16 @@
...
+    super(`${message} ${formatErrorUrl(code)}`);
   }
 }
 
+const formatErrorUrl = (code: ErrorCode) => {
+  return `https://functionless.org/docs/error-codes/#${code.messageText
+    .toLowerCase()
+    .replace(/\s+/g, "-")}`;
+};
+
```
Comment (sam-goodwin) [inline src/error-code.ts:15]: There is similar logic in `validate.ts`. Consolidate?

**Review by sam-goodwin — CHANGES_REQUESTED** [review]

File: test-app/tsconfig.json [hunk 2]
```diff
@@ -19,6 +19,7 @@
     "noUnusedLocals": true,
     "noUnusedParameters": true,
     "resolveJsonModule": true,
+    "skipLibCheck": true,
```
Comment (sam-goodwin) [inline test-app/tsconfig.json:22]: what does this do?

Comment (thantos) [reply test-app/tsconfig.json:22]: Test app was failing on some node deps. just tells TS not to check TS in all of the node modules.

File: src/error-code.ts [hunk 3]
```diff
@@ -3,9 +3,24 @@
...
   constructor(readonly code: ErrorCode, message: string) {
-    super(message);
+    super(formatErrorMessage(code, message));
   }
 }
+const baseUrl = process.env.FUNCTIONLESS_LOCAL
+  ? `http://localhost:3000`
+  : `https://functionless.org`;
+
+export const formatErrorMessage = (code: ErrorCode, messageText?: string) => `${
```
Comment (sam-goodwin) [inline src/error-code.ts:13]: Document these functions?

File: src/error-code.ts [hunk 4]
```diff
@@ -3,9 +3,24 @@
...
+  messageText ?? code.messageText
+}
+
+${formatErrorUrl(code)}`;
+
+const formatErrorUrl = (code: ErrorCode) => {
```
Comment (sam-goodwin) [inline src/error-code.ts:19]: Document

**Review by sam-goodwin — CHANGES_REQUESTED** [review]

**Review by sam-goodwin — APPROVED** [review]

**Pull request merged**